sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -41,10 +59,108 @@ not rely on APIs defined in this specification.*
 
 == Introduction
 
-This extension introduces an interface that enables a lazy execution and easy replay of a kernel graph by separating
-its definition and execution.
-
-== Feature test macro
+Through the use of command groups SYCL is already able to create a DAG of kernel
+execution at runtime, as a command group object defines a set of requisites
+(edges) which must be satisfied for kernels (nodes) to be executed. However,
+because command-group submission is tied to execution on the queue, without
+having a prior construction step before starting execution, optimization
+opportunities are missed from the runtime not knowing the complete dependency
+graph ahead of execution.
+
+The following benefits would become possible if the user could define a
+dependency graph to the SYCL runtime prior to execution:
+
+* Reduction in runtime overhead by only submitting a single graph object, rather
+  than many individual commands.
+
+* Enable more work to be done offline, in particular producing a graph ahead of
+  time allows for improved performance at runtime from reduced overhead.
+
+* Unlock DMA hardware features through graph analysis by the runtime.
+
+* Whole graph optimizations become available, including but not limited to:
+** Kernel fusion/fission.
+** Inter-node memory reuse from data staying resident on device.
+** Identification of the peak intermediate output memory requirement, used for
+   more optimal memory allocation.
+
+As well as benefits to the SYCL runtime, there are also advantages to the user
+developing SYCL applications, as repetitive workloads no longer have to
+redundantly issue the same sequence of commands. Instead, a graph is only
+constructed once and submitted for execution as many times as is necessary, only
+changing the data in input buffers or USM allocations. For applications from
+specific domains, such as machine learning, where the same command group pattern
+is run repeatedly for different inputs, this is particularly useful.
+
+=== Requirements
+
+In order to achieve the goals described in previous sections, the following
+requirements were considered:
+
+1. Ability to update inputs/outputs of the graph between submissions, without
+   changing the overall graph structure.
+2. Enable low effort porting of existing applications to use the extension.
+3. Profiling, debugging, and tracing functionality at the granularity of graph
+   nodes.
+4. Integrate sub-graphs (previously constructed graphs) when constructing a new
+   graph.
+5. Support the USM model of memory as well as buffer model.
+6. Compatible with other SYCL extensions and features, e.g kernel fusion &
```

Comment:
I guess it makes sense to re-read the whole draft before submitting it to improve its language.

Reply:
Updated this, the main problem I have with re-reading the whole draft is that I've spent so long working on this document that I find it quite hard to read objectively without projecting what I think it says. Also, as a native English speaker I've never had to learn grammar rules properly 😄 

If you have time to do that pass Julian it'd be great, otherwise for now I'd tend towards relying on external reviewers coming to the spec for the first time commenting when the language isn't clear to them.